framework/aster-frame/src/vm/kspace.rs

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Kernel memory space management.
+
+use align_ext::AlignExt;
+use spin::Once;
+
+use super::page_table::PageTableConstsTrait;
+use crate::{
+    arch::mm::{PageTableConsts, PageTableEntry},
+    vm::{
+        page_table::{page_walk, KernelMode, MapCachePolicy, MapProperty, PageTable},
+        space::VmPerm,
+        MemoryRegionType, Paddr, Vaddr, PAGE_SIZE,
+    },
+};
+
+/// The base address of the linear mapping of all physical
+/// memory in the kernel address space.
+pub(crate) const LINEAR_MAPPING_BASE_VADDR: Vaddr = 0xffff_8000_0000_0000;
+
+/// The kernel code is linear mapped to this address.
+///
+/// FIXME: This offset should be randomly chosen by the loader or the
+/// boot compatibility layer. But we disabled it because the framework
+/// doesn't support relocatable kernel yet.
+pub fn kernel_loaded_offset() -> usize {
+    0xffff_ffff_8000_0000
+}
+
+pub fn vaddr_to_paddr(va: Vaddr) -> Option<Paddr> {
```

Comment:
The exact behaviors of this method is not clear to me as it lack of Rust doc. The function returns an `Option<_>`, which seems to suggest **only a valid virtual address will have its physical address returned**. But apparently, there exists some virtual addresses that fall within `LINEAR_MAPPING_BASE_VADDR..=kernel_loaded_offset()`, but are not backed by any main memory or MMIO regions.

I think this API is fundamentally flawed. Translating a virtual address through a direct mapping or a page table satisfies two different use cases. Using a single API to address two distinct needs with two different costs is not wise.

I checked all current usage of this function. There are three usage: two of them in `io_mem.rs`, one in `virtio_queue.rs`. The usage in `virtio_queue` will be soon removed (as the usage itself is flawed). The two in `io_mem.rs` can only rely on the direct mapping, not the page. So it can use another API that only translate virtual addresses that fall within the direct mapping.

This problem is not introduced by this PR. But fixing it can be easily done in this PR, preferably in a separate commit. I suggest marking this API deprecated and providing well-defined alternatives.